pkg/tsdb/elasticsearch/client/search_request.go

```diff
@@ -92,6 +92,19 @@ func (b *SearchRequestBuilder) AddDocValueField(field string) *SearchRequestBuil
 	return b
 }
 
+// Add highlights to the search request for log queries
+func (b *SearchRequestBuilder) AddHighlight() *SearchRequestBuilder {
+	b.customProps["highlight"] = map[string]interface{}{
+		"fields": map[string]interface{}{
+			"*": map[string]interface{}{},
+		},
+		"pre_tags":      []string{"@HIGHLIGHT@"},
+		"post_tags":     []string{"@/HIGHLIGHT@"},
+		"fragment_size": 2147483647,
```

Comment:
what an interesting number! 😄 

i looked it up, and it is the largest 32bit integer, when dealing with signed integers (meaning integers that can do negative numbers too, they go from `-2147483648` to `2147483647`)

Reply:
Fyi https://www.agilytic.be/blog/tech-talk-elasticsearch-highlighting-kibana if you are interested in more reading on this.